cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -278,12 +282,16 @@ public void awaitTerminated() throws ExecutionException {
   public void awaitTerminated(long timeout, TimeUnit timeoutUnit) throws TimeoutException, ExecutionException {
     boolean interrupted = false;
     try {
+      long remainingNanos = timeoutUnit.toNanos(timeout);
+      long end = System.nanoTime() + remainingNanos;
+
       while (true) {
         try {
-          completion.get(timeout, timeoutUnit);
+          completion.get(remainingNanos, TimeUnit.NANOSECONDS);
```

Comment:
Shouldn't we check if remainingNanos is greater than zero? I don't think the docs specify what happens if the supplied value is a negative number.

Reply:
if `remainingNano` is < 0, it throws `TimeoutException`, which is what we wanted.